### PR TITLE
API Proposal: Updated APIs for channel member refactor

### PR DESCRIPTION
**Pull request by jwilander**
Proposal for API changes in mattermost/platform#4106

### CODE REVIEW

File: source/channels.yaml [hunk 1]
```diff
@@ -142,3 +142,58 @@
...
+          required: true
+          type: string
+        - name: user_id
+          in: path
+          description: The user ID
+          required: true
+          type: string
+      responses:
+        '200':
+          description: The channel stats
```
Comment (enahum) [inline source/channels.yaml]: should this be `channel member` instead of channel stats?

File: source/definitions.yaml [hunk 2]
```diff
@@ -105,6 +105,14 @@ definitions:
       creator_id:
         type: string
 
+  ChannelStats:
+    type: object
+    properties:
+      id:
```
Comment (enahum) [inline source/definitions.yaml]: I prefer to be more consistent here maybe as with ChannelMember where `id` is actually `channel_id`?

File: source/users.yaml [hunk 3]
```diff
@@ -114,8 +115,101 @@
...
+      description: Get a map of sanitized user objects that are on the team but not in the channel.
+      parameters:
+        - name: team_id
+          in: path
+          description: The team ID
+          required: true
+          type: string
+        - name: channel_id
+          in: path
+          description: The channel ID
```
Comment (enahum) [inline source/users.yaml]: `The channel ID the users are not part of` ? to be more explicit

File: source/definitions.yaml (hunk 2 above)
Comment (jwilander) [inline source/definitions.yaml]: That was from the legacy stuff, but yes that's a good idea. I'll update

File: source/users.yaml (hunk 3 above)
Comment (jwilander) [inline source/users.yaml]: Agreed, will update

File: source/channels.yaml (hunk 1 above)
Comment (jwilander) [inline source/channels.yaml]: Oops, that's right

**Comment by jwilander** [comment]
@enahum Changes made, take a look

**Comment by jwilander** [comment]
I made some updates to the new APIs:
* Changed `/users/profiles_by_ids` to `/users/ids` to be more consistent with REST and to be consistent with future similar APIs.
* Changed `/teams/{team_id}/channels/{channel_id}/member/{user_id}` to '/teams/{team_id}/channels/{channel_id}/members/{user_id}' to keep consistent resource name for `members` APIs

**Comment by enahum** [comment]
+1

**Pull request merged**